x-pack/plugins/stack_alerts/server/rule_types/geo_containment/executor.ts

```diff
@@ -98,7 +98,10 @@ export async function executor({
         windowEnd,
       });
       if (context) {
-        recoveredAlert.setContext(context);
+        services.alertsClient?.setAlertData({
```

Comment:
Can we replace the recovered alerts loop with 

```
for (const recoveredAlert of services.alertsClient?.getRecoveredAlerts()) {
  const recoveredAlertId = recoveredAlert.alert.getId();
...
}
```
instead of using `services.alertFactory.done()`?